### ISSUE TITLE
Create UV Editor Window; or better curved UV algorithm.

### ISSUE THREAD

**Issue by Henry00IS** [sabresaurus/SabreCSG#67]
Curved brushes are incredibly difficult to UV map seamlessly with the surface tools currently available in SabreCSG. AutoUV can't generate satisfying UVs. It would be nice to have a UV editor window that can deal with these types of scenarios or introduce another AutoUV that can handle curvature better.

![image](https://user-images.githubusercontent.com/7905726/37824632-8abd56fc-2e8d-11e8-9c39-308f6ea08827.png)

![uvreason1](https://user-images.githubusercontent.com/7905726/37828196-909cda4a-2e9a-11e8-86b2-a2fd792b0b33.png)

![uvreason2](https://user-images.githubusercontent.com/7905726/37828199-9363b38e-2e9a-11e8-9942-f72f923b49e5.png)

Edit: **Research Notes**
* Cube Projection appears to be a simple yet powerful candidate that can solve this.
* Stackoverflow: [Automatically generating UV coordinates algorithms](https://stackoverflow.com/questions/18578943/automatically-generating-uv-coordinates-algorithms)
* Blender: [ED_uvedit_unwrap_cube_project](https://github.com/dfelinto/blender/blob/master/source/blender/editors/uvedit/uvedit_unwrap_ops.c)

**Comment by vertxxyz** [comment]
Doesn't the Follow Face uving option (ie. CTRL+SHIFT+LMB) UV this just
fine? Or does it produce different results to this?

**Comment by Henry00IS** [comment]
I love how you always... lol, not again! I thought that was some awkward feature that didn't let you select the wrong face or something. I added this feature [to the wiki](https://github.com/sabresaurus/SabreCSG/wiki/Brush-Tools-Face#follow-face) accordingly. I can't thank you enough, your comments are truly invaluable. 😁

I think that follow face is able to produce the same results yes. I should make it more user friendly though, can't be bothered to click a hundred times. The algorithm I suggested (an edit that you probably didn't see in your email, the Cube Projection Algorithm) would take care of this too:

![Cube Projection in Blender](https://user-images.githubusercontent.com/7905726/37827971-c5dcfb50-2e99-11e8-9ac7-57a1eedb0fa5.png)

Perhaps if follow face is as user friendly as selecting faces is these days, there's no need to implement another algorithm.
